File: 5_Sparse_Check/sparse_matrix.py
def is_sparse(matrix):
    zeroes = sum(row.count(0) for row in matrix)
    return zeroes > (len(matrix) * len(matrix[0])) / 2

r, c = map(int, input().split())
matrix = [list(map(int, input().split())) for _ in range(r)]
if is_sparse(matrix):
    print("Yes, Sparse Matrix")
else:
    print("No, Not Sparse")

File: 5_Sparse_Check/sparse_matrix.c
#include <stdio.h>
#include <stdbool.h>
bool isSparse(int matrix[100][100], int r, int c) {
    int count = 0;
    for (int i = 0; i < r; i++)
        for (int j = 0; j < c; j++)
            if (matrix[i][j] == 0) count++;
    return count > (r * c) / 2;
}
int main() {
    int r, c;
    scanf("%d %d", &r, &c);
    int matrix[100][100];
    for (int i = 0; i < r; i++)
        for (int j = 0; j < c; j++)
            scanf("%d", &matrix[i][j]);
    if (isSparse(matrix, r, c))
        printf("Yes, Sparse Matrix\n");
    else
        printf("No, Not Sparse\n");
    return 0;
}
